rafthttp/snapshot_sender.go

```diff
@@ -104,7 +105,9 @@ func (s *snapshotSender) send(merged snap.Message) {
 // post posts the given request.
 // It returns nil when request is sent out and processed successfully.
 func (s *snapshotSender) post(req *http.Request) (err error) {
-	cancel := httputil.RequestCanceler(req)
+	ctx, cancel := context.WithCancel(context.Background())
+	req = req.WithContext(ctx)
+	defer cancel()
```

Comment:
@xiang90 for that we will need to change the design a bit since we cannot use defer directly there. For this reason govmt is not giving the error there. Using defer will not cause the code to work properly. So we will need to move the go routine code to i.e. L124 to L128 outside go routine and the code from L132 to L156 inside the go routine. Is that fine with you?